src/Cache/QueryCacheProfile.php

```diff
@@ -22,12 +22,12 @@ class QueryCacheProfile
 {
     public function __construct(
         private readonly int $lifetime = 0,
-        private readonly ?string $cacheKey = null,
-        private readonly ?CacheItemPoolInterface $resultCache = null
+        private readonly string|null $cacheKey = null,
+        private readonly CacheItemPoolInterface|null $resultCache = null,
```

Comment:
I don't believe that union types should be enforced over the nullable ones. We discussed it recently in another thread.

Reply:
Can you please open an issue on the CS repo? I think that's were the issue needs to be discussed.

Reply:
Done: https://github.com/doctrine/coding-standard/issues/288.